### ISSUE TITLE
adaptive timeout for VP check if there is a progress

### ISSUE THREAD

**Issue by mkaczanowski** [ChorusOne/blazar#3]
# TL;DR
Blazar VP post check waits X seconds (timeout) for the consensus. If after X seconds there is no 2/3+1, blazar fails postcheck.

We know that upgrades can last a very long time, so a better approach IMHO is to dynamically extend the timeout when there is a progress.
For instance, say the timeout is 10 minutes:
1. post check starts
2. after 5 minutes post check detects VP increase
3. the post check timeout resets
4. another increase happens
5. post check timeout resets
6. finalize
a) post check timeouts after 10 minutes after last reset if there is no consensus
b) post check completes when there is consensus, this could be after 15 minutes
